src/ophyd_async/core/detector.py

```diff
@@ -171,7 +171,10 @@ async def trigger(self) -> None:
         indices_written = await self.writer.get_indices_written()
         written_status = await self.controller.arm(DetectorTrigger.internal, num=1)
         await written_status
-        await self.writer.wait_for_index(indices_written + 1)
+        await asyncio.wait(
+            [self.writer.wait_for_index(indices_written + 1)],
+            timeout=DEFAULT_TIMEOUT,
+        )
```

Comment:
Should this be `wait` or `wait_for`?

Looking at the api, the only differences are:
1. wait accepts an iterable of futures/tasks, but wait_for accepts a single awaitable.
2. wait doesnt cancel the futures when a timeout occurs, but wait_for does.

I think it's neater to use wait_for personally.
4.

Reply:
solved in https://github.com/bluesky/ophyd-async/pull/55